src/data_sources/model/DataRecord.ts

```diff
@@ -0,0 +1,155 @@
+   * @returns {String} - The path of the record.
+   * @name getPath
+   * @example
+   * const pathRecord = record.getPath();
+   * // e.g., 'SOURCE_ID.record1'
+   * const pathRecord2 = record.getPath('myProp');
+   * // e.g., 'SOURCE_ID.record1.myProp'
+   */
+  getPath(prop?: string, opts: { useIndex?: boolean } = {}) {
+    const { dataSource, id, index } = this;
+    const dsId = dataSource.id;
+    const suffix = prop ? `.${prop}` : '';
+    return `${dsId}.${opts.useIndex ? index : id}${suffix}`;
+  }
+
+  /**
+   * Get both ID-based and index-based paths of the record.
+   * Returns an array containing the paths using both ID and index.
+   *
+   * @param {String} [prop] - Optional property name to include in the paths.
+   * @returns {Array<String>} - An array of paths.
+   * @name getPaths
+   * @example
+   * const paths = record.getPaths();
+   * // e.g., ['SOURCE_ID.record1', 'SOURCE_ID.0']
+   */
+  getPaths(prop?: string) {
+    return [this.getPath(prop), this.getPath(prop, { useIndex: true })];
+  }
+
+  /**
+   * Trigger a change event for the record.
+   * Optionally, include a property name to trigger a change event for a specific property.
+   *
+   * @param {String} [prop] - Optional property name to trigger a change event for a specific property.
+   * @name triggerChange
+   */
+  triggerChange(prop?: string) {
+    const { dataSource, em } = this;
+    const data = { dataSource, dataRecord: this };
+    const paths = this.getPaths(prop);
+    paths.forEach((path) => em.trigger(`${DataSourcesEvents.path}:${path}`, { ...data, path }));
+  }
+
+  /**
+   * Set a property on the record, optionally using transformers.
+   * If transformers are defined for the record, they will be applied to the value before setting it.
+   *
+   * @param {String|Object} attributeName - The name of the attribute to set, or an object of key-value pairs.
+   * @param {any} [value] - The value to set for the attribute.
+   * @param {Object} [options] - Options to apply when setting the attribute.
+   * @param {Boolean} [options.avoidTransformers] - If true, transformers will not be applied.
+   * @returns {DataRecord} - The instance of the DataRecord.
+   * @name set
+   * @example
+   * record.set('name', 'newValue');
+   * // Sets 'name' property to 'newValue'
+   */
+  set<A extends _StringKey<T>>(
+    attributeName: Partial<T> | A,
+    value?: SetOptions | T[A] | undefined,
+    options?: SetOptions | undefined,
+  ): this;
+  set(attributeName: unknown, value?: unknown, options?: SetOptions): DataRecord {
+    const onRecordSet = this.dataSource?.transformers?.onRecordSet;
+
+    const newValue =
+      options?.avoidTransformers || !onRecordSet
+        ? value
+        : onRecordSet({
+            id: this.id,
+            key: attributeName as string,
+            value,
+          });
+
+    super.set(attributeName as string, newValue, options);
+
+    return this;
+  }
```

Comment:
You know, I like how you're managing onRecordSet here, makes sense to me. 

Quite curious why onRecordGet is not working the same way. You're actually suggesting mutating the model in onRecordGet, right? Are we sure this is the right way? 🤔

ps: Backbone is actually supporting also multiple properties updates, eg. `record.set({ key1: '', key2: '' })`, IMHO we have to support it here 🥲

Reply:
@artf please see commit: 

* https://github.com/GrapesJS/grapesjs/pull/6018/commits/5dc734bd6dbcc46a9ad18341eaf94093bab586a9

Reply:
@artf see commit

I removed the following transformers: 

1. onRecordDelete
2. onRecordRead
3. onRecordSet

And now we only have one: `onRecordSetValue`. This is more than enough for someone to validate each entry and its called on record add and on record set, tested in `/tests/data_sources/transformers.ts`. I think this is a simpler API to start with.